src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -47,6 +47,12 @@ static inline uint32_t _ialpha(uint32_t c)
 }
 
 
+static inline uint32_t _grayvalue(uint32_t c)
+{
+    return ((((c&0xff)*19) + (((c>>8)&0xff)*183) + (((c>>16)&0xff)*54)) * (c >> 24) + 0xffff) >> 16; //(0.0721*B + 0.7154*G + 0.2125*R) * A
+}
```

Comment:
color is already alpha premultiplied, so you don't need to repeat that here

what if a color was build using _abgrJoin ?

Reply:
thanks. fixed by introducing surface->blender.grayValue function